common/primitives/timestamp/duration.go

```diff
@@ -31,9 +31,15 @@ import (
 	"google.golang.org/protobuf/types/known/durationpb"
 )
 
+// 250 years. Maximum representable golang time.Duration is approximately 290 years (INT64_MAX * time.Nanosecond)
+const maxAllowedDuration = 250 * 365 * 24 * time.Hour
+
 var (
 	errNegativeDuration = fmt.Errorf("negative duration")
 	errMismatchedSigns  = fmt.Errorf("duration has seconds and nanos with different signs")
+
+	maxSeconds = maxAllowedDuration.Nanoseconds() / 1e9
+	maxNanos   = int32(maxAllowedDuration.Nanoseconds() % 1e9)
```

Comment:
this is obviously always zero since an hour is a round number of seconds, can we just not make a constant?